### PR TITLE
Implementations of head_mut, get_mut, index_mut and improve docs

### PR DESCRIPTION
**Pull request by ehsanmok**
This PR adds some more convenient mutable counterparts of `head(), get() and index()` methods.

Also some minor improvement in docs related to [this](https://github.com/steveklabnik/indexlist/issues/13) and the already supported [into_iter](https://github.com/steveklabnik/indexlist/issues/4).

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -672,7 +786,6 @@ where
         };
 
         let removed = std::mem::replace(
-            // we know this index is valid thanks to the get on line 224
```
Comment (steveklabnik) [inline src/lib.rs:723]: is there a reason to remove these comments?

**Review by steveklabnik — COMMENTED** [review]
This looks fantastic! I have just one comment about some removed comments. Sorry this took me so long to review.

File: src/lib.rs (hunk 1 above)
Comment (steveklabnik) [reply src/lib.rs:723]: oops, i'm dumb. sorry!

**Review by steveklabnik — APPROVED** [review]

**Comment by steveklabnik** [comment]
Ah, this has a merge conflict now, can you rebase it? Sorry about that!

**Comment by ehsanmok** [comment]
@steveklabnik sure, no worries!

Also wanted to say, thank you for all your contributions to Rust. Without determinations like yours, we (community) wouldn't have such an awesome language to spend our time, enjoy and make a difference :heart:

**Comment by ehsanmok** [comment]
@steveklabnik ping.

**Comment by steveklabnik** [comment]
<3

**Pull request merged**